R/derive_joined.R

```diff
@@ -1003,31 +1006,66 @@ get_joined_data <- function(dataset,
     filter_if(filter_add) %>%
     ungroup()
 
-  # number observations of the input dataset and the additional dataset for
-  # relation of records, e.g., join_type = before|after, first_cond_lower,
-  # first_cond_upper
+  # number groups with respect to by_vars and order in the input dataset and the
+  # additional dataset for relation of records, e.g., join_type = before|after,
+  # first_cond_lower, first_cond_upper
   tmp_obs_nr_var_join <- NULL
   if (join_type != "all" || !is.null(first_cond_lower) || !is.null(first_cond_upper) ||
     !is.null(tmp_obs_nr_var)) {
     if (is.null(tmp_obs_nr_var)) {
       tmp_obs_nr_var <- get_new_tmp_var(dataset, prefix = "tmp_obs_nr")
       tmp_obs_nr_var_join <- paste0(as_name(tmp_obs_nr_var), ".join")
     }
-    data_add <- derive_var_obs_number(
-      data_add,
-      new_var = !!tmp_obs_nr_var,
-      by_vars = by_vars,
-      order = order,
-      check_type = check_type
-    )
+
+    if (check_type != "none") {
+      # check if order results in unique values in dataset and dataset_add
+      signal_duplicate_records(
+        dataset = data,
+        by_vars = c(by_vars, order),
+        msg = paste(
+          "Dataset {.arg dataset} contains duplicate records with respect to",
+          "{.var {replace_values_by_names(by_vars)}}"
+        ),
+        cnd_type = check_type
+      )
+      signal_duplicate_records(
+        dataset = data_add,
+        by_vars = c(by_vars, order),
+        msg = paste(
+          "Dataset {.arg dataset_add} contains duplicate records with respect to",
+          "{.var {replace_values_by_names(by_vars)}}"
+        ),
+        cnd_type = check_type
+      )
+    }
 
     data <- data %>%
-      mutate(!!!order) %>%
+      mutate(!!!order)
+
+    groups <- bind_rows(
+      select(data, !!!by_vars, !!!replace_values_by_names(order)),
+      select(data_add, !!!by_vars, !!!replace_values_by_names(order))
+    ) %>%
+      distinct() %>%
       derive_var_obs_number(
         new_var = !!tmp_obs_nr_var,
         by_vars = by_vars,
         order = order,
-        check_type = check_type
+        check_type = "none"
```

Comment:
just making sure that  `check_type = "none"` is intentional?  i think i understand why but just making sure!!

Reply:
Yes, it is intentional. Due to the `distinct()` call before, there can't be any duplicates. Thus I want to avoid spending time on checking it.